tools/isobuild/compiler-plugin.js

```diff
@@ -1394,8 +1397,19 @@ export class PackageSourceBatch {
     if (canCache) {
       LINKER_CACHE.set(cacheKey, ret);
       if (cacheFilename) {
+        const cacheKeySuffixRegex = new RegExp('_' + cacheKeySuffix, 'g');
+        const oldCacheFilename =
+          cacheFilename.replace(cacheKeySuffixRegex, '_*');
         // Write asynchronously.
-        Fiber(() => files.writeFileAtomically(cacheFilename, retAsJSON)).run();
+        Fiber(() => {
+          try {
+            files.rm_recursive(oldCacheFilename);
```

Comment:
This might be a good place to use [`files.rm_recursive_async`](https://github.com/meteor/meteor/blob/f0dd494c82244df79b544bdbf590ebecc0412189/tools/fs/files.js#L292), so we don't block writing the new `cacheFilename`.